shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -1804,6 +1814,7 @@ - (void)onOrientationPreferencesUpdated:(NSNotification*)notification {
 }
 
 - (void)performOrientationUpdate:(UIInterfaceOrientationMask)new_preferences {
+#if !APPLICATION_EXTENSION_API_ONLY
```

Comment:
Might be able to fall back to changing `UIWindowSceneGeometryPreferencesIOS` on `self.viewIfLoaded.window.windowScene` instead, if that works.